quinn-proto/src/cid_generator.rs

```diff
@@ -31,6 +31,33 @@ pub trait ConnectionIdGenerator: Send + Sync {
     fn cid_lifetime(&self) -> Option<Duration>;
 }
 
+/// Validates connection id in short header packet
+pub trait ConnectionIdValidator {
+    /// Validate and regconize a connection id from given buffer, return the length of connection id
+    /// or [`None`] if it is invalid
+    fn validate(&self, buffer: &[u8]) -> Option<usize>;
+}
+
+pub(crate) struct FixedLengthConnectionIdValidator {
+    expected_len: usize,
+}
+
+impl FixedLengthConnectionIdValidator {
+    pub(crate) fn new(expected_len: usize) -> Self {
+        Self { expected_len }
+    }
+}
+
+impl ConnectionIdValidator for FixedLengthConnectionIdValidator {
+    fn validate(&self, buffer: &[u8]) -> Option<usize> {
+        if buffer.len() >= self.expected_len {
```

Comment:
Nit: write this as `(buffer.len() >= self.expected_len).then_some(self.expected_len)`.